test/core.jl

```diff
@@ -5117,6 +5117,15 @@ f_isdefined_va(::T...) where {T} = @isdefined T
 @test !f_isdefined_va()
 @test f_isdefined_va(1, 2, 3)
 
+# @isdefined in a loop
+let a = []
+    for i = 1:2
+        push!(a, @isdefined(j))
+        j = 1
```

Comment:
possibly should mark this `local j`, so that it's clear what this is testing, and to make sure the insertion of another earlier test (or other random global state) doesn't invalidate this test